src/main/java/com/google/cloud/tools/appengine/cloudsdk/internal/process/StringBuilderProcessOutputLineListener.java

```diff
@@ -19,23 +19,23 @@
 import com.google.cloud.tools.appengine.cloudsdk.process.ProcessOutputLineListener;
 
 /**
- * A ProcessOutputLineListener that uses a StringBuilder to store the contents of all lines.
+ * A ProcessOutputLineListener that uses a StringBuffer to store the contents of all lines.
  */
 public class StringBuilderProcessOutputLineListener implements ProcessOutputLineListener {
 
-  private StringBuilder stringBuilder;
+  // This used to be a StringBuilder but that wasn't thread safe.
+  private final StringBuffer buffer = new StringBuffer();
 
   public StringBuilderProcessOutputLineListener() {
-    stringBuilder = new StringBuilder();
   }
 
   @Override
-  public void onOutputLine(String line) {
-    stringBuilder.append(line);
+  public synchronized void onOutputLine(String line) {
+    buffer.append(line);
```

Comment:
Hmm, this implementation has a flaw in that multiple lines will be catenated together without a newline. Doesn't matter for JSON or single-line output.  Might be worth adding a note to the class javadoc to this effect rather than trying to fix this.